File: 2023/06.29/정호조/공통부분문자열.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class 공통부분문자열_5582 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String str1 = br.readLine();
        String str2 = br.readLine();
        int[][] arr = new int[str1.length() + 1][str2.length() + 1];
        int ans = 0;

        for (int i = 1; i <= str1.length(); i++) {
            for (int j = 1; j <= str2.length(); j++) {
                if (str1.charAt(i - 1) == str2.charAt(j - 1)) {
                    arr[i][j] = arr[i - 1][j - 1] + 1;
                    ans = Math.max(ans, arr[i][j]);
                }
            }
        }

        System.out.println(ans);
    }
}

File: 2023/06.29/조수연/공통_부분_문자열.kt
import java.io.*
import kotlin.math.max

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {

    val bw = BufferedWriter(OutputStreamWriter(System.out))

    val str = readLine()
    val str2 = readLine()

    val arr = Array(str.length+1){IntArray(str2.length+1)}
    var length = 0

    for (i in 1..str.length){
        for (j in 1..str2.length){
            if (str[i-1] == str2[j-1]){ // 해당 위치에 문자가 서로 같을 때
                arr[i][j] = arr[i-1][j-1] + 1 // 해당 인덱스의 값은 그 전의 인덱스값 + 1 -> 현재 위치 -1의 위치에 문자 비교값 + 1
                length = max(length,arr[i][j])
            }
        }
    }

    bw.write("$length")

    bw.flush()
    bw.close()
}